pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -441,9 +441,12 @@ func (rq *Controller) Sync(ctx context.Context, discoveryFunc NamespacedResource
 			utilruntime.HandleError(err)
 
 			if discovery.IsGroupDiscoveryFailedError(err) && len(newResources) > 0 {
-				// In partial discovery cases, don't remove any existing informers, just add new ones
+				groupLookupFailures := err.(*discovery.ErrGroupDiscoveryFailed).Groups
+				// In partial discovery cases, don't remove any existing informers for resources in the failed groups, just add new ones
 				for k, v := range oldResources {
-					newResources[k] = v
+					if _, failed := groupLookupFailures[k.GroupVersion()]; failed {
```

Comment:
I'm confused - in your description you note we already made this change to the RQ controller.. but it looks like we are only just making it here?

We definitely _do_ want this change included - but I'm not sure how it relates to your comment:

> We already made this change to the resource quota controller in the past for similar reasons, and this PR narrows that retention to group/versions with discovery resolution errors and adds test coverage for it.